go/mysql/server.go

```diff
@@ -260,7 +261,10 @@ func (l *Listener) handle(conn net.Conn, connectionID uint32) {
 		c.sequence = 0
 		data, err := c.readEphemeralPacket()
 		if err != nil {
-			log.Errorf("Error reading packet from client %v: %v", c.ConnectionID, err)
+			// Don't log EOF errors. They cause too much spam.
+			if !strings.HasSuffix(err.Error(), "EOF") {
+				log.Errorf("Error reading packet from client %v: %v", c.ConnectionID, err)
+			}
```

Comment:
Let's change readEphemeralPacket to return io.EOF directly if it can't read enough, that way we don't do string compare here?